doc: initial moderation policy

jasnell opened the pull request:
@nodejs/tsc ... This is a proposed default moderation policy for the github organization. Please consider it a work in progress. The goal is for moderation actions to be as clear and transparent as possible.

Specific highlights:

* Collaborators must not moderate other Collaborators posts without giving them at least 24 hours to edit it themselves
* All moderation actions must be clearly documented
* Use of the private nodejs/moderation repo is defined

/cc @nodejs/collaborators ... I would appreciate feedback from collaborators on this to ensure it meets the needs of the community.

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,48 @@
+### Policy
+
+* All Posts are expected to respect the Node.js Code of Conduct.
+* Only Collaborators with commit rights to a given repository may Moderate Posts within that repository's issue tracker.
```
Does the TSC have power to supersede this?

jasnell replied:
It's more a practical ACL issue really. Not all collaborators have access rights to all repositories. The collaborator has to be granted write or admin access the repo in order to delete or modify comments. Also, all TSC members are automatically Owners of the Github repo and therefore have commits rights to every repo.

TheAlphaNerd replied:
so would this then be a bit of a moot point due to ACL? If you don't have commit rights you wouldn't be able to moderate a post

TheAlphaNerd replied:
although I guess codifying this makes it clear as opposed to assumed

jasnell replied:
Yep, I still think it's important to call out, however. There's likely going to be question of why only certain collaborators have the ability to moderate in certain repos

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,48 @@
+
+* All Posts are expected to respect the Node.js Code of Conduct.
+* Only Collaborators with commit rights to a given repository may Moderate Posts within that repository's issue tracker.
+* Collaborator's must not Moderate any Post authored by another Collaborator without first giving the author an opportunity of at least 24 hours to modify or remove the Post on their own.
```
If a post by a collaborator has something extremely offensive in it would it be permissible to moderate while properly documenting in nodejs/moderation?

This could happen even in good faith, for example an account being compromised.

jasnell replied:
Good question. Open to suggestions on that.

TheAlphaNerd replied:
perhaps
> Collaborator's must not Moderate any Post authored by another Collaborator that appears to be in good faith without first giving the author an opportunity of at least 24 hours to modify or remove the Post on their own.

jasnell replied:
Too hard to define what "in good faith" means here.

TheAlphaNerd replied:
maybe
> Collaborator's must not Moderate any Post authored by another Collaborator without first giving the author an opportunity of at least 24 hours to modify or remove the Post on their own. In extreme circumstances the TSC can be consulted to waive the 24 hour grace period

jasnell replied:
+1 .. that works for me

Starefossen replied:
At first I thought this was a bad suggestion since all collaborators are expected to follow the CoC and the CoC should apply equally to all members, collaborators or not. Then I re-read the CoC and found that is has a lot of «suggestions» (like the use of personal pronouns) which I think does not warrant an edit by another collaborator without notice.

jasnell replied:
Updated

jasnell replied:
@Starefossen ... exactly. The Code of Conduct, by default, takes a fairly prescriptive approach without trying to be heavy handed. I think this is a right approach.

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,48 @@
+ * A new post within the original thread, or
+ * A new issue within the private nodejs/moderation repository.
+* Explanations of Moderation actions for non-Collaborator Posts must be provided in:
+ * The original Post being modified (as the replacement content),
```
this can also be a cavaet at the end of the post

jasnell replied:
Updated

Starefossen replied:
Yes. What, why and by whom should be clearly stated at the bottom of the post when moderating. Openness is key.

Starefossen commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,48 @@
+
+### Terms
+
+* "Collaborator" refers to any individual member or "outside collaborator" of the Node.js Github Organization.
```
I think this should be a collaborator on the given project, not the organisation as a whole. Most collaborators does not have the possibility to edit posts in other repositories.

jasnell replied:
That's dealt with further down in the policy.

TheAlphaNerd replied:
If I am not mistaken this is dealt with in line 21

--> https://github.com/nodejs/TSC/pull/16/files#diff-ba1bbe10e98d34ecb91c62040189c025R21

ashleygwilliams commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,48 @@
+
+* All Posts are expected to respect the Node.js Code of Conduct.
+* Only Collaborators with commit rights to a given repository may Moderate Posts within that repository's issue tracker.
+* Collaborator's must not Moderate any Post authored by another Collaborator without first giving the author an opportunity of at least 24 hours to modify or remove the Post on their own. In extreme circumstances the TSC can be consulted to waive the 24 hour grace period.
```
- what does "not moderate" mean? does this not mean commenting that a piece of content violates the code of conduct?
- what does extreme mean here?
- are there exceptions for clear trolling? (e.g. the node/inclusivity group's 8chan/4chan situation)
- when does the 24hrs start? from post or fro notification that it needs to be edited/changed? who comments that something needs to be edited/changed?

generally, i am curious what the motivation is here. why not just delete/moderate the comment and the collaborator can repost something more constructive/appropriate?

TheAlphaNerd replied:
@ashleygwilliams this is specifically for cases when moderating a comment made by a collaborator in the nodejs org. Not for general traffic which is below

TheAlphaNerd replied:
I'm under the impression that this is meant to encourage good faith discussion between collaborators and to give the potentially offending individual the opportunity to rectify their actions.

ashleygwilliams replied:
yup i totally get that and think that is cool, but the specifics seems more complicated than on first glance. i figure being clear is important.

jasnell replied:
* Moderate is defined further up in the document. Posting a new comment that calls out any code of conduct violation is separate
* Extreme = obvious gross violations of the code of conduct
* The 8chan/4chan situation involved non-Collaborator posts, which are addressed separately from this bullet point.
* The 24hrs starts from the request to edit/change. I can clarify this.

The motivation is transparency and collaboration while giving fellow collaborators the benefit of the doubt. Note that moderation of non-Collaborators is handled differently.

TheAlphaNerd replied:
@ashleygwilliams the language here is definitely tricky and a couple variation have been tossed about. I agree that being clear is super important. Any suggestions on how to clear it up?

Starefossen commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,48 @@
+ * A new issue within the private nodejs/moderation repository.
+* Any Collaborator who habitually authors Posts that must be Moderated will be subject to possible removal from the Node.js Github Organization. Such action can only be taken through normal TSC motion and vote.
+* Moderation of Posts authored by non-Collaborators may result in those non-Collaborators being Banned from further participation in the Node.js Github Organization for an indefinite period of time.
+* Only a TSC member may Ban an individual from the Node.js Github Organization.
```
Until there is a Moderation WG?

TheAlphaNerd replied:
I'm getting the impression that this moderation policy will be giving groups their own autonomy for moderation, and thus would not require an over arching moderation wg. correct me if I am mistaken

jasnell replied:
I don't believe a Moderation WG is necessary under this policy.

Starefossen commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,48 @@
+ * The original Post being modified (as the replacement content),
+ * A new post within the original thread, or
+ * A new issue within the private nodejs/moderation repository.
+* Any Collaborator who habitually authors Posts that must be Moderated will be subject to possible removal from the Node.js Github Organization. Such action can only be taken through normal TSC motion and vote.
```
Collaborators can be excluded from their respective projects by the governing working group according to their governance policy as normal, or does this have to go through the TSC?

ashleygwilliams commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,48 @@
+
+### Modifications to This Policy
+
+Modifications to this policy are made through normal TSC motion and vote. Any Collaborator may submit a PR proposing changes to this policy. Those PRs must be labeled using the `tsc-agenda` label.
```
link to details of this process maybe?

ashleygwilliams commented:
this generally looks really great. a couple of thoughts:

- might be easier to read if you separated policies for collaborators vs non collaborators
- a separate, specific section on how to report things for moderation

thanks for putting this together! it's def something we need.

ashleygwilliams commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,48 @@
+
+By default, this policy applies to all repositories under the Node.js Github Organization and all Node.js Working Groups.
+
+Individual Working Groups and Top Level Projects chartered by the TSC may adopt an alternative Moderation Policy for any repository under their stewardship so long as:
```
really love this. thanks for including it.

jbergstroem replied:
Can repositories be stewarded by multiple working groups? If so, which policy would then apply?

jasnell replied:
Good question. That's an edge case that's not really addressed anywhere in the governance documents. Perhaps we need to be able to say that every repo has a primary owner which it either the TSC, CTC or a WG.

jasnell replied:
@jbergstroem ... ok new commit deals with this

jasnell commented:
@TheAlphaNerd, @ashleygwilliams , @Starefossen ... just added a commit with multiple updates.

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,80 @@
+* All Posts are expected to respect the Node.js Code of Conduct.
+* Only Collaborators with commit rights to a given repository may Moderate Posts within that repository's issue tracker.
+* Only a TSC member may Ban an individual from the Node.js Github Organization.
+* Any individual Banned from the Node.js Github Organization will be recommended for exclusion from any Node.js Foundation sponsored event or activity.
```
:+1:

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,80 @@
+
+#### Collaborator Posts
+
+* Collaborator's must not Moderate any Post authored by another Collaborator without first giving the author an opportunity of at least 24 hours (from the time of the initial request) to modify or remove the Post on their own.
```
I don't know that ```an opportunity``` is necessary. Reads clearer to me as 

> Collaborator's must not Moderate any Post authored by another Collaborator without first giving the author at least 24 hours (from the time of the initial request) to modify or remove the Post on their own.

One thing though is we have no clarity about how the collaborators are supposed to communicate a concern

jasnell replied:
See Line 52 :-)

jasnell replied:
Will remove "an opportunity"

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,80 @@
+ * A new post within the original thread, or
+ * A new issue within the private nodejs/moderation repository.
+* Any Collaborator who habitually authors Posts that must be Moderated will be subject to possible removal from the Node.js Github Organization. Such action can only be taken through normal TSC motion and vote.
+
```
I really like how this list has tightened up. It leaves a lot less to the imagination and clarifies process.

We may want to make a note regarding archiving the original post. That way if there is a disagreement an archive could be used to avoid [gaslighting](https://en.wikipedia.org/wiki/Gaslighting)

jasnell replied:
Good point. I will add text indicating that requests for moderation should include a screenshot of the Post

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,80 @@
+* Explanations of Moderation actions for non-Collaborator Posts must be provided in:
+ * The original Post being modified (as replacement or appended content),
+ * A new post within the original thread, or
+ * A new issue within the private nodejs/moderation repository.
```
I think it is important in this case for there to still be some level of transparency of moderated content. Simply moderating without making mention to the moderated content can give ammunition to those claiming that they are the discussion is being manipulated. That being said it should obviously not be mandatory, as publicly documenting the moderation could make you a target.

Truthfully we probably should have a moderation guide to go along with this policy. The guide can be used to train, whereas I see the purpose of this document to set process.

jasnell replied:
As best practice, perhaps, but doing so could trigger unwanted attention. The most typical case for non-collaborator moderation will be dealing with trolls which should just be handled as quietly and as quickly as possible. Note that this section doesn't even make explaining the moderation a requirement (says "should" not "must")

TheAlphaNerd replied:
Indeed, probably better to document as a best practice. What I would love to see is "dealing with trolls 101"

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,80 @@
+
+#### Requesting Moderation
+
+Requests to Moderate a Post are only necessary when the Post in question was written by a Collaborator.
```
This may make sense to live higher in the document and specify various cases where requesting moderation is necessary. This could be a great place to outline our moderation / reporting policy for non-collaborators

jasnell replied:
You mean move the entire section up higher?

TheAlphaNerd replied:
If the section is to include information about how a non collaborator requests moderation I think it should be the first thing on the list.

jasnell replied:
+1 ... will move it up, not quite the first thing but higher...

TheAlphaNerd replied:
I was assuming below definitions

misterdjules commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,80 @@
+Requesting Moderation of a Post can be accomplished in one of five ways:
+
+* Private communication between the individuals involved,
+* Via the `report@io.js` email address,
```
We use `security@nodejs.org` for the security mailing list, anything `@io.js` seems like it would be confusing. Can we do `report@nodejs.org`?

jasnell replied:
This is currently a technical limitation. See the discussion on https://github.com/nodejs/email/issues/5

misterdjules replied:
Commented in that issue.

jasnell replied:
FWIW, i'd very much prefer `@nodejs.org`

thefourtheye replied:
Isn't `security@nodejs.org` for security vulnerabilities?

jasnell replied:
@thefourtheye ... @misterdjules was using that as an illustration for instance. The report email alias should ideally use the `@nodejs.org` domain and not `@io.js`

thefourtheye replied:
Oops, sorry. Need some coffee..

jasnell replied:
Me too. It's been a long day.

misterdjules commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,80 @@
+
+### Modifications to This Policy
+
+Modifications to this policy are made through normal [TSC motion and vote](https://github.com/nodejs/TSC/blob/master/TSC-Charter.md#section-8-voting). Any Collaborator may submit a PR proposing changes to this policy. Those PRs must be labeled using the `tsc-agenda` label. Including a mention to `@nodejs/tsc` can be used to call the issue to TSC's attention.
```
Does that apply to the initial version of this document? It seems like it should.

jasnell replied:
Yep

jasnell commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,95 @@
+* Only a TSC member may Ban an individual from the Node.js Github Organization.
+* Any individual Banned from the Node.js Github Organization will be recommended for exclusion from any Node.js Foundation sponsored event or activity.
+
+#### Requesting Moderation
```
Hmm... still not quite happy with the location of this section... likely will move it up a bit higher still

jasnell commented:
Ok, additional edits made. PTAL

jasnell commented:
Ok, I think this is largely complete. @nodejs/tsc ... would appreciate a look and feedback when y'all have a moment

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,100 @@
+
+Requesting Moderation of a Post can be accomplished in one of five ways:
+
+* Private communication between the individuals involved,
```
This now seems odd to be the first option suggested if these are the rules for non collaborators. It may be considered insensitive to suggest someone goes to the person who needs to be moderated

jasnell replied:
Good point. I'll move that to the end of the list

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,100 @@
+### Terms
+
+* "Collaborator" refers to any individual member or "outside collaborator" of the Node.js Github Organization.
+* "Post" refers to the content of any issue, pull request, comment or wiki page.
```
Should we define "Community Member" here as well?

jasnell replied:
Well, we don't actually use that term anywhere in here so it may be a bit confusing. "community" is extremely difficult to define.

TheAlphaNerd replied:
I think having tldr instructions at the top alleviates this concern
as well as my concern in https://github.com/nodejs/TSC/pull/16/files#r45420728

thefourtheye commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,100 @@
+
+#### Collaborator Posts
+
+* Collaborator's must not Moderate any Post authored by another Collaborator without first giving the author at least 24 hours (from the time of the initial request) to modify or remove the Post on their own.
```
Collaborators must not ...

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,100 @@
+## Moderation policy
+
```
We should add a very quick TLDR here something akin to

> if you are not a member of the node.js org please send moderation request to  report@nodejs.org or ... (other communication methods)

or something to that effect

thefourtheye commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,100 @@
+
+#### Non-Collaborator Posts
+
+* Posts authored by non-Collaborator's are always subject to immediate Moderation by any Collaborator if the content is intentionally disruptive or in violation of the Node.js Code of Conduct.
```
non-Collaborators are always....

jasnell replied:
sigh... me and my cursed muscle memory...

thefourtheye commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,100 @@
+
+### Consideration of Intent
+
+Before Moderating a Post, Collaborator's should carefully consider the possible intent of the author. It may be that the author has simply made an error or is not familiar with the Code of Conduct; or it may be that cultural differences exist or that the author is unaware that certain content is considered inappropriate. Unless the offense or the intent to disrupt, annoy or harass is obvious, an  ...
```
Collaborators should carefully...

jasnell commented:
@TheAlphaNerd @misterdjules ... PTAL. Updated the intro with a quick tl/dr statement, added a table of contents and switched to using the `report@nodejs.org` alias.

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+
+External public venues or social media services such as Twitter should never be used to request Moderation or to discuss the details of a Moderation request.
+
+Multiple moderation requests for the same Post are unnecessary.
```
I don't know that we need to be explicit here. Feels like a different tone than the rest of the document

jasnell replied:
For which part? The multiple moderation requests? I want to avoid a situation where someone begins blasting requests everywhere (emails to tsc members, posts to the public issue, posts to the moderation repo, etc).

TheAlphaNerd replied:
To me not sending multiple requests is a common sense kind of thing. The individual who will abuse this will do so whether or not this is in the document.

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+
+Requests should contain as much information and context as possible, including the URL and a screenshot of the Post in question. Screenshots may be modified to obscure obscene or offensive content.
+
+External public venues or social media services such as Twitter should never be used to request Moderation or to discuss the details of a Moderation request.
```
I think we want to be very careful about the language here. I think it is fair to say not to use those venues to request Moderation, but it feels out of scope of this document to limit discussion

jasnell replied:
The goal here is not to limit discussion but to avoid situations where someone purposefully goes out to draw unhelpful outside attention that inflames the issue even further.

jasnell replied:
For instance, imagine a tweet, "Hey! So-n-so at Node just said my post about Such-n-Such is inappropriate! Go tell them how you feel!", etc

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+
+Note that Moderating non-Collaborator posts can often lead to retaliation or escalation of inappropriate behavior by the individual whose post is being Moderated. This is true primarily of individuals whose intent is to harass, disrupt or annoy individual members of the community. In such cases, it is best to handle the Moderation as quickly and as quietly as possible without drawing any further  ...
+
+### Consideration of Intent
```
Do you think this section about intent should live near the top of the document. It could be a nice reminder before someone gets bogged down with process

jasnell replied:
Wouldn't hurt

thefourtheye commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+* Via private email to individual TSC members,
+* Via a new Post in the same thread as the Post being requested for Moderation,
+* Via a new Post in the private nodejs/moderation repository, or
+* Private communication between the individuals involved,
```
How will we know if someone is really authroized to moderate a post if the communication is private?

Starefossen replied:
Nit picking, but I there should be a `.` instead of a `,` at the end of this line.

jasnell replied:
This line may just need to be removed. The author of the post can always
edit it or delete it.
On Nov 19, 2015 9:36 PM, "thefourtheye" <notifications@github.com> wrote:

> In Moderation-Policy.md
> <https://github.com/nodejs/TSC/pull/16#discussion_r45437426>:
>
> > +### Requesting Moderation
> > +
> > +Anyone may request Moderation of a Post.
> > +
> > +Moderation requests are only necessary when:
> > +
> > +* The Post in question was authored by a Collaborator, or
> > +* The request is originating from a non-Collaborator.
> > +
> > +Requesting Moderation of a Post can be accomplished in one of five ways:
> > +
> > +* Via the `report@node.js` email address,
> > +* Via private email to individual TSC members,
> > +* Via a new Post in the same thread as the Post being requested for Moderation,
> > +* Via a new Post in the private nodejs/moderation repository, or
> > +* Private communication between the individuals involved,
>
> How will we know if someone is really authroized to moderate a post if the
> communication is private?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/TSC/pull/16/files#r45437426>.
>

jasnell replied:
Also, TSC members always have the authorization to moderate posts because they automatically become Owners in the GitHub organization (giving them access to every repository)

thefourtheye commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+
+### Consideration of Intent
+
+Before Moderating or a Post, Collaborators should carefully consider the possible intent of the author. It may be that the author has simply made an error or is not familiar with the [Code of Conduct](https://github.com/nodejs/node/blob/master/CODE_OF_CONDUCT.md); or it may be that cultural differences exist or that the author is unaware that certain content is considered inappropriate. Unless th ...
```
Before moderating a post...

thefourtheye commented:
Should we change all Github to GitHub, as the company's actual name is written as the latter?

thefourtheye commented:
Also few words are capitalised even if they are in the middle of the sentence. For example, almost everywhere it is either Moderating, Moderator or Moderation.

jasnell commented:
Those are capitalized because they are defined terms in the Terms section
above. It's a convention that's sometimes used in formal standards but
could be changed.
On Nov 19, 2015 9:42 PM, "thefourtheye" <notifications@github.com> wrote:

> Also few words are capitalised even if they are in the middle of the
> sentence. For example, almost everywhere it is either Moderating, Moderator
> or Moderation.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/TSC/pull/16#issuecomment-158284640>.
>

jasnell commented:
@thefourtheye @Starefossen ... fixed those couple of issues and made a couple additional edits.

Fishrock123 commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,116 @@
+Moderation requests are only necessary when:
+ 
+* The Post in question was authored by a Collaborator, or
+* The request is originating from a non-Collaborator.
```
I don't get this bit?

jasnell replied:
May need to work on the wording here then. Essentially, any Collaborator can moderate any post by a non-Collaborator at any time without first warning or asking; but if the Post was written by a a Collaborator, the author should be asked to change it first. Or, if the request is coming from someone who is not a collaborator (and therefore doesn't have the ability to moderate the posts on their own) they would need to submit a request.

Fishrock123 replied:
Yeah that doesn't convey here.

Fishrock123 commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,116 @@
+
+Use of the `report@node.js` email address, or private email to individual TSC members, is appropriate when the individual requesting the Moderation does not feel comfortable directly or publicly making the request. All emails sent to the `report@node.js` address are currently forwarded to all members of the TSC. [[ED NOTE: This has not yet been set up]]
+
+When a request is sent by email either to the `report@node.js` or directly to a TSC member, a new issue detailing the request must be created in the private nodejs/moderation repository. The identity of the individual submitting the request may be omitted from that issue.
```
Should be omitted unless specified by the reporter?

jasnell replied:
+1 .. will update that

Fishrock123 commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,116 @@
+
+### Consideration of Intent
+
+Before Moderating a Post, Collaborators should carefully consider the possible intent of the author. It may be that the author has simply made an error or is not familiar with the [Code of Conduct](https://GitHub.com/nodejs/node/blob/master/CODE_OF_CONDUCT.md); or it may be that cultural differences exist or that the author is unaware that certain content is considered inappropriate. Unless the o ...
```
Being unfamiliar with the CoC shouldn't be an excuse not to moderate posts when necessary though.

jasnell replied:
Yes, but the author should at least first be given an opportunity to correct the error themselves.

Fishrock123 commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,116 @@
+ * A new issue within the private nodejs/moderation repository.
+* Moderation of Posts authored by non-Collaborators may result in those non-Collaborators being Banned from further participation in the Node.js GitHub Organization for an indefinite period of time.
+
+Note that Moderating non-Collaborator posts can often lead to retaliation or escalation of inappropriate behavior by the individual whose post is being Moderated. This is true primarily of individuals whose intent is to harass, disrupt or annoy individual members of the community. In such cases, it is best to handle the Moderation as quickly and as quietly as possible without drawing any further  ...
```
I suggest we change offensive and/or troll issue titles to `-`, since it gives those issues even less visibility in searches.

We should also change the content to something similar.

jasnell replied:
Good point. I should mention titles also.

jasnell commented:
@Fishrock123 .. ok, pushed some edits to update your comments. PTAL

orangemocha commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+
+### Terms
+
+* "Collaborator" refers to any individual member or "outside collaborator" of the Node.js GitHub Organization.
```
This definition is not clear to me. Is there a less ambiguous way to define it? Is it people listed as collaborators on the readme / members of the nodejs/collaborators team? And what is an "outside collaborator"?

jasnell replied:
It's any public/private member of the Node.js GitHub Organization as listed in https://github.com/orgs/nodejs/people and any "outside collaborator" as listed in https://github.com/orgs/nodejs/outside-collaborators. "Outside Collaborator" is a GitHub term used to describe individuals who have been granted access to a repository but are not members of the Organization

jasnell replied:
I can clarify that in the text

jasnell replied:
@misterdjules updated.

orangemocha commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+Use of the `report@node.js` email address -- or private email to individual TSC members -- is appropriate when the individual requesting the Moderation does not feel comfortable directly or publicly making the request. All emails sent to the `report@node.js` address are currently forwarded to all members of the TSC. [[ED NOTE: This has not yet been set up]]
+
+When a request is sent by email to the `report@node.js` (or directly to a TSC member) a new issue detailing the request must be created in the private nodejs/moderation repository. The identity of the individual submitting the request should be omitted from that issue unless specified by the reporter.
+
```
Knit: "unless specified by the reporter" is slightly ambiguous and could be interpreted as "unless the **identity** is specified by the reported".

jasnell replied:
Updated!

orangemocha commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+When a request is sent by email to the `report@node.js` (or directly to a TSC member) a new issue detailing the request must be created in the private nodejs/moderation repository. The identity of the individual submitting the request should be omitted from that issue unless specified by the reporter.
+
+Requests should contain as much information and context as possible, including the URL and a screenshot of the Post in question. Screenshots should be modified to obscure obscene or offensive content.
+
```
> Screenshots should be modified to obscure obscene or offensive content.

I am not sure this is necessary or advisable. Don't the moderators need to be able to see the content in order to exercise their judgment? I know it's not a fun job...

jasnell replied:
Yes, but if the screen shots are accessible to all collaborators in nodejs/moderation we need to be careful. The content itself may need to be obscured for legal reasons. I originally had this as a "may" but changed it to a "should". I can change that back to "may"

orangemocha commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+Collaborators should never discuss the specific details of a Moderation request in any public forum or any social media service outside of the Node.js GitHub Organization.
+
+Note that Collaborators may Moderate non-Collaborator Posts at any time without submitting an initial request (see: [Non-Collaborator Posts](#non-collaborator-posts)).
+
```
If Collaborators can moderate without submitting the request, and it's the expected way for them, it would be better to make it clear at the beginning of this section rather than leave it at end.

jasnell replied:
Moved higher. Not quite to the beginning but not at the end.

jasnell commented:
@targos ... yeah, that's why I had originally left those links out. Not quite sure how to be more exact about the definition of Collaborators tho without having those in. Perhaps Collaborator is simply any person with Repository access?

orangemocha commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+### Consideration of Intent
+
+Before Moderating a Post, Collaborators should carefully consider the possible intent of the author. It may be that the author has simply made an error or is not yet familiar with the [Code of Conduct](https://GitHub.com/nodejs/node/blob/master/CODE_OF_CONDUCT.md); or it may be that cultural differences exist, or that the author is unaware that certain content is considered inappropriate. In such ...
+
```
I got to reading this section and realized that the doc doesn't specify (yet) what constitute grounds for moderating a post, or requesting moderation. Is it violation of code of conduct? It would be good to clarify it early in the doc.

jasnell replied:
Good point. Will add.

orangemocha commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+
+Note, however, that unfamiliarity with the [Code of Conduct](https://GitHub.com/nodejs/node/blob/master/CODE_OF_CONDUCT.md) does not excuse a Post from Moderation.
+
+### Policy
```
Consider changing the name of this section, or incorporating its content in another section. "Policy" is also the name of the document, so a section named the same is slightly confusing.

jasnell replied:
How about "Guidelines and Requirements" ?

orangemocha replied:
Sure

orangemocha commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+#### Collaborator Posts
+
+* Collaborators must not Moderate any Post authored by another Collaborator without first giving the author at least 24 hours (from the time of the initial request) to modify or remove the Post on their own.
+* If the author of the Post disagrees that Moderation is required, the matter can be escalated to the TSC by creating a new issue (or using an existing one) within the nodejs/moderation repository with the label `tsc-agenda`. In such cases, the TSC will serve as the final arbiter.
```
The `tsc-agenda` label is used to tag things for meetings agenda. Mentioning nodejs/tsc would guarantee a faster response. Setting the tsc-agenda label could be optional IMO.

jasnell replied:
+1 Will update. Actually going to move TSC escalation out to a separate section to make that bit clearer

orangemocha commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,113 @@
+* When Moderating non-Collaborator Posts, the moderating Collaborator should:
+ * Explain the justification for Moderating the post, and
+ * Identify all changes made to the Post.
+* If an explanation of a Moderation action for a non-Collaborator Post is provided, it must be provided in:
```
Maybe change `must` to `should`? The `If` implies that an explanation is not mandatory, so the set of methods for it is probably best express as a recommendation

jasnell replied:
+1 will update

orangemocha commented:
Thanks for putting this together. Generally looks great! Left some comments.

One more question: does quoting the original context, which may contain offensive language, for the purpose of moderation also consist a violation of the code of conduct? Would it be useful to clarify if and in which way / purpose / contexts (eg private moderation repo) this may be admissible?

jasnell commented:
@targos ... I updated the definition on Collaborator again, can you take a look. I think it's an improvement

targos commented:
:+1: It's better now.

jasnell commented:
@orangemocha ... ok, included a clarification that quoting is not a violation but that discretion should be used.

sam-github commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,129 @@
+* All Posts are expected to respect the [Node.js Code of Conduct](https://GitHub.com/nodejs/node/blob/master/CODE_OF_CONDUCT.md).
+* Only Collaborators with commit rights to a given repository may Moderate Posts within that repository's issue tracker.
+* The TSC serves as the final arbiter for all Moderation issues (see: [Escalation to the TSC](#escalation-to-the-tsc)).
+* Only a TSC member may Ban an individual from the Node.js GitHub Organization.
```
One, single, TSC member, or only the TSC as a whole?

sam-github commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,129 @@
+
+All Moderation-related decisions follow the normal [TSC motion and voting process](https://GitHub.com/nodejs/TSC/blob/master/TSC-Charter.md#section-8-voting).
+
+TSC members directly involved in a Moderation issue -- as either the requester or author of the Post in question -- are expected to excuse themselves from any decisions required to resolve the issue.
```
"the requestor of moderation" is what what "requestor" means, I think, and would be more explicit (as opposed to the PR "requestor", where the PR is the one that someone else made a moderated comment on).

brendanashworth commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,129 @@
+* "Collaborator" refers to any individual with configured access permissions to any Node.js GitHub organization repository. See [GitHub's access permissions documentation](https://help.github.com/articles/what-are-the-different-access-permissions/) for more information.
+* "TSC" refers to the [Node.js Technical Steering Committee](https://github.com/nodejs/node#tsc-technical-steering-committee).
+* "Post" refers to the content and titles of any issue, pull request, comment or wiki page.
+* "Moderate" refers to the act of modifying the content and title of, or deleting, any Post.
```
Is this intended to also include smaller, good-faith edits like typos? Or would those be excluded?

TheAlphaNerd replied:
I think this is intended to include any and all communication or contributions within the nodejs organization.

jasnell replied:
Personally I'd prefer that no one make any edits in someone else's posts
unless they're specifically for moderation, good faith or otherwise.
On Nov 23, 2015 11:00 PM, "Brendan Ashworth" <notifications@github.com>
wrote:

> In Moderation-Policy.md
> <https://github.com/nodejs/TSC/pull/16#discussion_r45702032>:
>
> > +Individual Working Groups and Top Level Projects chartered by the TSC may adopt an alternative Moderation Policy for any repository under their stewardship so long as:
> > +* The Moderation Policy is openly documented as part of the Working Group or Project Charter and;
> > +* Includes provisions for clearly and openly documenting Moderation actions taken.
> > +
> > +If a particular repository can be considered to fall under the stewardship of multiple Working Groups (or Top Level Projects) that have adopted different Moderation Policies, they can choose to either:
> > +* Decide between themselves which Moderation Policy will be in effect for the repository in question, or
> > +* Ask the TSC to determine which Moderation Policy should apply.
> > +
> > +Any alternative Moderation Policy used for a given repository must be included in the root directory of the repository using the `Moderation-Policy.md` filename. If a repository does not contain a `Moderation-Policy.md` file, then this default policy applies.
> > +
> > +### Terms
> > +
> > +* "Collaborator" refers to any individual with configured access permissions to any Node.js GitHub organization repository. See [GitHub's access permissions documentation](https://help.github.com/articles/what-are-the-different-access-permissions/) for more information.
> > +* "TSC" refers to the [Node.js Technical Steering Committee](https://github.com/nodejs/node#tsc-technical-steering-committee).
> > +* "Post" refers to the content and titles of any issue, pull request, comment or wiki page.
> > +* "Moderate" refers to the act of modifying the content and title of, or deleting, any Post.
>
> Is this intended to also include smaller, good-faith edits like typos? Or
> would those be excluded?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/TSC/pull/16/files#r45702032>.
>

bnoordhuis replied:
Counterpoint: I frequently fix up comments where people dump unformatted logs into an issue.

jasnell replied:
@bnoordhuis ... Hmm.. Perhaps adding a clause that such edits and their
intent need to be documented with a quick note?
On Nov 25, 2015 5:14 AM, "Ben Noordhuis" <notifications@github.com> wrote:

> In Moderation-Policy.md
> <https://github.com/nodejs/TSC/pull/16#discussion_r45864615>:
>
> > +Individual Working Groups and Top Level Projects chartered by the TSC may adopt an alternative Moderation Policy for any repository under their stewardship so long as:
> > +* The Moderation Policy is openly documented as part of the Working Group or Project Charter and;
> > +* Includes provisions for clearly and openly documenting Moderation actions taken.
> > +
> > +If a particular repository can be considered to fall under the stewardship of multiple Working Groups (or Top Level Projects) that have adopted different Moderation Policies, they can choose to either:
> > +* Decide between themselves which Moderation Policy will be in effect for the repository in question, or
> > +* Ask the TSC to determine which Moderation Policy should apply.
> > +
> > +Any alternative Moderation Policy used for a given repository must be included in the root directory of the repository using the `Moderation-Policy.md` filename. If a repository does not contain a `Moderation-Policy.md` file, then this default policy applies.
> > +
> > +### Terms
> > +
> > +* "Collaborator" refers to any individual with configured access permissions to any Node.js GitHub organization repository. See [GitHub's access permissions documentation](https://help.github.com/articles/what-are-the-different-access-permissions/) for more information.
> > +* "TSC" refers to the [Node.js Technical Steering Committee](https://github.com/nodejs/node#tsc-technical-steering-committee).
> > +* "Post" refers to the content and titles of any issue, pull request, comment or wiki page.
> > +* "Moderate" refers to the act of modifying the content and title of, or deleting, any Post.
>
> Counterpoint: I frequently fix up comments where people dump unformatted
> logs into an issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/TSC/pull/16/files#r45864615>.
>

bnoordhuis replied:
Sounds reasonable.

jasnell commented:
@mikeal ... added some initial language regarding trolling. basically, the language allows the TSC to define trolling behavior. We may need to be more specific. Let me know what you think.

@bnoordhuis @brendanashworth ... added allowance for smaller formatting edits and typo fixes. PTAL.

jasnell commented:
@nodejs/tsc @nodejs/ctc @nodejs/inclusivity ... ok, I think this is ready to go. Can I get another review, particularly from fellow TSC members?

Trott commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,139 @@
+* "Moderate" refers to the act of modifying the content and title of, or deleting, any Post for the purpose of correcting or addressing Code of Conduct violations.
+* "Ban" refers to the act of blocking an individual GitHub account from any further participation in the Node.js GitHub Organization.
+* "Requester" refers to an individual requesting Moderation on a Post.
+
```
Nittiest of nits: Italics is better than quotation marks for words-as-words like in this situation. So:

* _Collaborator_ refers to any individual...
* _TSC_ refers to...
* _Post_ refers to...

orangemocha commented:
LGTM

bnoordhuis commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,139 @@
+
+Anyone may request Moderation of a Post. Requesting Moderation of a Post can be accomplished in one of four ways:
+
+* Via the `report@node.js` email address,
```
Maybe make this a mailto: link?

bnoordhuis replied:
Also, shouldn't that be `report@nodejs.org`, here and below?

jasnell replied:
Sigh. Yep, fixing

bnoordhuis commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,139 @@
+
+### Escalation to the TSC
+
+Any Moderation issue or dispute can be escalated to the TSC by "mentioning" `@nodejs/tsc` in the body of a GitHub issue either in the original thread or in the private nodejs/moderation repository. Optionally, the `tsc-agenda` label may be attached to the issue to request that it be added to the TSC meeting agenda.
```
At mentions of groups only work when you're a member of the node.js organization, I think.  If the intent of this section is that only collaborators escalate, that should be mentioned explicitly because it isn't clear from context (to me anyway.)

bnoordhuis commented:
Is the section on trolling necessary?  It seems a little superfluous to me.

Fishrock123 commented:
@bnoordhuis yes, because sometimes it can be hard to identify so we should address it directly.

bnoordhuis commented:
Okay, but why explicitly call out trolling?  That's just asking for language lawyering about what constitutes trolling.  I'd phrase it as e.g. "The TSC has final authority in deciding what is disruptive behavior and how to moderate it."

(Not in those exact words but you get the idea.)

jasnell commented:
+1 I like that revised wording
On Dec 1, 2015 1:53 PM, "Ben Noordhuis" <notifications@github.com> wrote:

> Okay, but why explicitly call out trolling? That's just asking for
> language lawyering about what constitutes trolling. I'd phrase it as e.g.
> "The TSC has final authority in deciding what is disruptive behavior and
> how to moderate it."
>
> (Not in those exact words but you get the idea.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/TSC/pull/16#issuecomment-161107474>.
>

mikeal commented:
> Okay, but why explicitly call out trolling? That's just asking for language lawyering about what constitutes trolling. I'd phrase it as e.g. "The TSC has final authority in deciding what is disruptive behavior and how to moderate it."

So, I don't think we should identify trolling *behavior* I think that we should identify trolling *accounts*. The reason for this is that a lot of modern trolling is done not by comments that would violate the CoC but instead by intentionally trying to waste peoples time and discourage consensus. This is easily identifiable as the accounts are usually created no more than 24 hours before the comment.

I agree that defining "trolling behavior" is problematic and potentially far too subjective. Any account with actual history should fall within the CoC and doesn't need any other consideration but moderators do need the ability to more aggressively moderate throwaway accounts.

jasnell commented:
@mikeal ... agreed that the current proposed language isn't ideal. Given that I had added this after you and I discussed it, it would be awesome if you could help with some alternative text ;-)

mikeal commented:
@jasnell I know, I've just been slammed lately and was also sick for almost a week. Will get this on my TODO list though :)

TheAlphaNerd commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,139 @@
+
+Note that Moderating non-Collaborator posts can often lead to retaliation or escalation of inappropriate behavior by the individual whose post is being Moderated. This is true primarily of individuals whose intent is to harass, disrupt or annoy individual members of the community. In such cases, it is best to handle the Moderation as quickly and as quietly as possible without drawing any further  ...
+
+#### Trolling
```
If we were to remove this paragraph we could probably replace it with a single line to the above section regarding Non Collaborator posts

* In the case where the offending account appears to have been created with no intention to collaborate in good faith swift actions may be taken without following the above mentioned procedures including
  * removing posts
  *  banning accounts
  *  reporting accounts to github

jasnell replied:
this is heading in the right direction but it needn't be limited to the creator of Posts but also to requesters. There are occasions where a troll will submit requests for moderation that ought to be ignored.

TheAlphaNerd replied:
maybe add it to guidelines rather than posts?

* In the case where an account that is in violation of the coc or moderation rules appears to have been created with no intention to collaborate in good faith swift actions may be taken without following the above mentioned procedures including

    * removing posts
    * banning accounts
    * reporting accounts to github

TheAlphaNerd commented:
@mikeal / @jasnell I took a quick stab at some copy around throw aways... let me know your thoughts

jasnell commented:
Pushed a revision. @bnoordhuis / @TheAlphaNerd / @mikeal ... PTAL

bnoordhuis commented on Moderation-Policy.md:
```diff
@@ -0,0 +1,138 @@
+
+Any Moderation issue or dispute can be escalated to the TSC by "mentioning" `@nodejs/tsc` in the body of a GitHub issue either in the original thread or in the private nodejs/moderation repository. Optionally, the `tsc-agenda` label may be attached to the issue to request that it be added to the TSC meeting agenda.
+
+(Note that using the `@nodejs/tsc` syntax to mention the TSC in the body of a request only will only work if the author of the Post is currently a member of the Node.js GitHub organization.)
```
Only one only will do. :-)

bnoordhuis commented:
LGTM

jasnell commented:
@nodejs/tsc @nodejs/ctc ... I'm going to give this another 24 hours before landing.

nebrius commented:
LGTM too. Thanks for the work @jasnell, this is awesome! :)

jasnell commented:
Landed in f1c83ad

sup commented:
woot!

TheAlphaNerd commented:
!!!

nebrius commented:
🎉
